app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt

```diff
@@ -38,10 +41,22 @@ class RecipeRepository(private val db: FirebaseFirestore) {
    * with their IDs, and then stores it in Firestore under the recipes collection.
    *
    * @param recipe The Recipe object to be added to Firestore.
+   * @param context The context of the calling activity.
    * @param onSuccess A callback invoked upon successful addition of the recipe.
    * @param onFailure A callback invoked upon failure to add the recipe, with an exception.
    */
-  fun addRecipe(recipe: Recipe, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+  fun addRecipe(
+      recipe: Recipe,
+      context: Context = FirebaseFirestore.getInstance().app.applicationContext,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    // Check if the user is offline
+    if (!isNetworkAvailable(context)) {
+      Log.d("addRecipe", "Offline mode: Cannot add recipe")
+      return
```

Comment:
`isNetworkAvailable` is a clever implementation, but why are you not calling onFailure before returning ?

Reply:
If I call onFailure it would throw exceptions in the logs, that's not what we want since this is an intended functionality.
However, I agree that the Toasts are stacking up, that's because I'm displaying it whenever the method isNetworkAvailable, is called and returns false, maybe I should display the Toast somewhere else.

Reply:
No, onFailure does not throw an exception, it's a function defined by the caller.
Hence it can be used to do pattern matching on the exception and display information in a more friendly way.

Not calling onFailure could break the app for feature that need a answer, such as generatedRecipe as we are waiting for an answer with a "loading",  yet if nothing happen, the app may be stuck in this loading state  forever....

However, I don't think such a loading has been implemented yet, that's why I did not require change.

Reply:
So you want me to call the method's `onFailure` each time the network connection is checked and unavailable ?

Reply:
> So you want me to call the method's `onFailure` each time the network connection is checked and unavailable ?

It would be the ideal but I think the current implementation work for now. 
It's the best-way without spending too much time checking all the caller ^^